src/main/scala/com/advancedtelematic/director/db/SetMultiTargets.scala

```diff
@@ -72,10 +72,18 @@ class SetMultiTargets()(implicit messageBusPublisher: MessageBusPublisher) exten
     dbAct.transactionally
   }
 
+  def getMultiTargetUpdates(namespace: Namespace, updateId: UpdateId)(implicit db: Database, ec: ExecutionContext): Future[MultiTargetUpdateRequest] = {
+    multiTargetUpdatesRepository
+      .fetch(updateId, namespace)
+      .map(_.map(mtuRow => mtuRow.hardwareId -> TargetUpdateRequest(mtuRow.fromTarget, mtuRow.toTarget, mtuRow.targetFormat, mtuRow.generateDiff)))
+      .map(_.toMap)
+      .map(MultiTargetUpdateRequest)
+  }
```

Comment:
This class should still handle only *Row data types (see other methods, they use that) and the conversion to request should only be done on the http level, in this case on the resource

Reply:
Oh I see what you mean. I've updated now.